### PR TITLE
Add a student id check

### PR DESCRIPTION
**Pull request by w-lfchen**
This pr aims to add a method to the build tasks that tries to catch students who enter their matriculation number in the `studentId` field, fail the build and inform them about their mistake.

### CODE REVIEW

File: src/main/kotlin/org/tudalgo/algomate/extension/SubmissionExtension.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import org.gradle.kotlin.dsl.property
 import javax.inject.Inject
 
 abstract class SubmissionExtension @Inject constructor(
-    objectFactory: ObjectFactory,
+    objectFactory: ObjectFactory
```
[the remark this answers is not in the document]
Comment (w-lfchen) [reply src/main/kotlin/org/tudalgo/algomate/extension/SubmissionExtension.kt]: I blame the formatter or something

File: src/main/kotlin/org/tudalgo/algomate/AlgoMatePlugin.kt [hunk 2]
```diff
@@ -92,6 +94,18 @@ class AlgoMatePlugin : Plugin<Project> {
                 sourceCompatibility = "17"
                 targetCompatibility = "17"
             }
+            withType<SubmissionWriteInfoTask> {
+                doFirst {
+                    if (!studentId.get().matches(".*[a-zA-Z].*".toRegex())) {
```
[the remark this answers is not in the document]
Comment (w-lfchen) [reply src/main/kotlin/org/tudalgo/algomate/AlgoMatePlugin.kt:99]: added in e50c01f

**Comment by Rdeisenroth** [comment]
Hmm could we not go for a more exact regex, like `[a-z]{2}\d{2}[a-z]{4}`?

See https://regex101.com/r/p0Irh8/1

**Comment by w-lfchen** [comment]
The initial point of this was to simply ensure that no matriculation numbers are entered. I'm not opposed to updating it, but is there a guarantee that all TU-IDs have this format?

**Comment by alexstaeding** [comment]
A more specific regex would, in fact, be incorrect. I have seen valid TU-IDs with only one letter (some old ones)

**Review by alexstaeding — APPROVED** [review]
LGTM, but I would still like to test it first before merging

**Comment by Rdeisenroth** [comment]
So, uhm did someone test it? 👀

**Review by Rdeisenroth — APPROVED** [review]
Works

**Pull request merged**